lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -229,4 +229,168 @@ public function get_styles_for_block( $block_metadata ) {
+	 *                       For example:
+	 *                       'classes' => array(
+	 *                         '.has-$slug-color'            => 'color',
+	 *                         '.has-$slug-background-color' => 'background-color',
+	 *                         '.has-$slug-border-color'     => 'border-color',
+	 *                       )
+	 * - properties       => array of CSS properties to be used by kses to
+	 *                       validate the content of each preset
+	 *                       by means of the remove_insecure_properties method.
+	 */
+	const PRESETS_METADATA = array(
+		array(
+			'path'              => array( 'color', 'palette' ),
+			'prevent_override'  => array( 'color', 'defaultPalette' ),
+			'use_default_names' => false,
+			'value_key'         => 'color',
+			'css_vars'          => '--wp--preset--color--$slug',
+			'classes'           => array(
+				'.has-$slug-color'            => 'color',
+				'.has-$slug-background-color' => 'background-color',
+				'.has-$slug-border-color'     => 'border-color',
+			),
+			'properties'        => array( 'color', 'background-color', 'border-color' ),
+		),
+		array(
+			'path'              => array( 'color', 'gradients' ),
+			'prevent_override'  => array( 'color', 'defaultGradients' ),
+			'use_default_names' => false,
+			'value_key'         => 'gradient',
+			'css_vars'          => '--wp--preset--gradient--$slug',
+			'classes'           => array( '.has-$slug-gradient-background' => 'background' ),
+			'properties'        => array( 'background' ),
+		),
+		array(
+			'path'              => array( 'color', 'duotone' ),
+			'prevent_override'  => array( 'color', 'defaultDuotone' ),
+			'use_default_names' => false,
+			'value_func'        => 'gutenberg_get_duotone_filter_property',
+			'css_vars'          => '--wp--preset--duotone--$slug',
+			'classes'           => array(),
+			'properties'        => array( 'filter' ),
+		),
+		array(
+			'path'              => array( 'typography', 'fontSizes' ),
+			'prevent_override'  => false,
+			'use_default_names' => true,
+			'value_key'         => 'size',
+			'css_vars'          => '--wp--preset--font-size--$slug',
+			'classes'           => array( '.has-$slug-font-size' => 'font-size' ),
+			'properties'        => array( 'font-size' ),
+		),
+		array(
+			'path'              => array( 'typography', 'fontFamilies' ),
+			'prevent_override'  => false,
+			'use_default_names' => false,
+			'value_key'         => 'fontFamily',
+			'css_vars'          => '--wp--preset--font-family--$slug',
+			'classes'           => array( '.has-$slug-font-family' => 'font-family' ),
+			'properties'        => array( 'font-family' ),
+		),
+		array(
+			'path'              => array( 'spacing', 'spacingSizes' ),
+			'prevent_override'  => false,
+			'use_default_names' => true,
+			'value_key'         => 'size',
+			'css_vars'          => '--wp--preset--spacing-size--$slug',
+			'classes'           => array( '.has-$slug-spacing-size' => 'spacing-size' ),
```

Comment:
I think we have a problem with the configuration here for the CSS class generation. This will assign the CSS var value to a CSS property called `spacing-size`. Shouldn't it be `padding`, `margin`, or `gap`? 

Being able to assign it to a relevant CSS property might cause a bit more bloat in terms of the number of generated classnames but as this is, it generates invalid CSS.

Reply:
Good 👀 ! That was staring me right in the face [above](https://github.com/WordPress/gutenberg/pull/41527#issuecomment-1149319835)!  🤦

Reply:
Good question. I was wondering if we even need to have classes for the spacing presets? Ultimately, the spacing for blocks that opt in to the `padding`, `margin`, and `blockGap` will be output directly via the style engine and layout support, rather than using the classname to assign the value to a particular property. For example, in global styles if the Group block is set to to use the CSS var, padding might be output as:

```
.wp-block-group {
  padding: var(--wp--preset--spacing-size--10);
}
```

@glendaviesnz is the intention for classnames to reflect the current state, or to be used to apply styles? (I wasn't too sure)

Reply:
> @glendaviesnz is the intention for classnames to reflect the current state, or to be used to apply styles? (I wasn't too sure)

Good question, haven't nailed that one down yet. I might just remove the classnames from this PR and just have this one focus on the theme.json settings and the css var properties.

Reply:
@aaronrobertshaw  - I have removed the class generation for now - we can decided if we need that and how to handle it in a later PR

Reply:
Thanks, I'll re-test shortly. 

At a quick glance, should it just be an empty array rather than deleted entirely? Like the duotone metadata.

Reply:
> should it just be an empty array rather than deleted entirely? Like the duotone metadata.

yeh, it probably should - done

Reply:
Something to consider for the generated classes – how would they work with directional values? If a group block has 10 vertical padding and 30 vertical padding then `--spacing size` wouldn't clearly apply. It might be better to just use the values directly in utility classes.